tests/tests/swfs/avm1/global_swf5_6_7_8_9/test.as

```diff
@@ -0,0 +1,151 @@
+
+  trace("  s1.global.__proto__          === s2.global.__proto__:          " + (s1.global.__proto__          === s2.global.__proto__));
+  trace("  s1.global2.__proto__         === s2.global2.__proto__:         " + (s1.global2.__proto__         === s2.global2.__proto__));
+  trace("  s1.anObject.__proto__        === s2.anObject.__proto__:        " + (s1.anObject.__proto__        === s2.anObject.__proto__));
+  trace("  s1.anArray.__proto__         === s2.anArray.__proto__:         " + (s1.anArray.__proto__         === s2.anArray.__proto__));
+  trace("  s1.aFunction.__proto__       === s2.aFunction.__proto__:       " + (s1.aFunction.__proto__       === s2.aFunction.__proto__));
+  trace("  s1.anObjectClass.__proto__   === s2.anObjectClass.__proto__:   " + (s1.anObjectClass.__proto__   === s2.anObjectClass.__proto__));
+  trace("  s1.aMovieClipClass.__proto__ === s2.aMovieClipClass.__proto__: " + (s1.aMovieClipClass.__proto__ === s2.aMovieClipClass.__proto__));
+  trace("  s1.aBooleanClass.__proto__   === s2.aBooleanClass.__proto__:   " + (s1.aBooleanClass.__proto__   === s2.aBooleanClass.__proto__));
+
+  trace("  s1.anObject instanceof s2.anObjectClass:   " + (s1.anObject instanceof s2.anObjectClass));
+}
+
+
+var mc5 = createEmptyMovieClip("child5", 2);
+var mc6 = createEmptyMovieClip("child6", 3);
+var mc7 = createEmptyMovieClip("child7", 4);
+var mc8 = createEmptyMovieClip("child8", 5);
+var mc9 = createEmptyMovieClip("child9", 6);
+
+var loader5 = new MovieClipLoader();
+loader5.loadClip("child5.swf", mc5);
+
+var loader6 = new MovieClipLoader();
+loader6.loadClip("child6.swf", mc6);
+
+var loader7 = new MovieClipLoader();
+loader7.loadClip("child7.swf", mc7);
+
+var loader8 = new MovieClipLoader();
+loader8.loadClip("child8.swf", mc8);
+
+var loader9 = new MovieClipLoader();
+loader9.loadClip("child9.swf", mc9);
+
+var toLoad = 5;
+var listener = {
+  onLoadInit: function() {
+    if (--toLoad == 0) {
+      test(mc5, "SWF5");
+      test(mc6, "SWF6");
+      test(mc7, "SWF7");
+      test(mc8, "SWF8");
+      test(mc9, "SWF9");
+      compare(mc5, mc6, "SWF5", "SWF6");
+      compare(mc5, mc7, "SWF5", "SWF7");
+      compare(mc5, mc8, "SWF5", "SWF8");
+      compare(mc5, mc9, "SWF5", "SWF9");
+      compare(mc6, mc5, "SWF6", "SWF5");
+      compare(mc6, mc7, "SWF6", "SWF7");
+      compare(mc6, mc8, "SWF6", "SWF8");
+      compare(mc6, mc9, "SWF6", "SWF9");
+      compare(mc7, mc5, "SWF7", "SWF5");
+      compare(mc7, mc6, "SWF7", "SWF6");
+      compare(mc7, mc8, "SWF7", "SWF8");
+      compare(mc7, mc9, "SWF7", "SWF9");
+      compare(mc8, mc5, "SWF8", "SWF5");
+      compare(mc8, mc6, "SWF8", "SWF6");
+      compare(mc8, mc7, "SWF8", "SWF7");
+      compare(mc8, mc9, "SWF8", "SWF9");
+      compare(mc9, mc5, "SWF9", "SWF5");
+      compare(mc9, mc6, "SWF9", "SWF6");
+      compare(mc9, mc7, "SWF9", "SWF7");
+      compare(mc9, mc8, "SWF9", "SWF8");
```

Comment:
Half of these are redundant; if `compare(x, y)` and `compare(y, x)` start giving different outputs we'll have a lot more test breakage besides!

Reply:
Not all comparisons are commutative, I would have to separate them into commutative and non commutative

Reply:
Let's just merge it as-is, I don't see any downside of including those redundant checks, it's not a lot of assertions and the test is fast anyway